File: baekjoon/단계별로 풀어보기/반복문/25304_영수증.py
# https://www.acmicpc.net/problem/25304

from sys import stdin

input = stdin.readline
X = int(input())
for _ in range(int(input())):
    a, b = map(int, input().split())
    X -= a * b
print('No' if X else 'Yes')

File: C#/Baekjoon/반복문/25304_영수증.cs
//https://www.acmicpc.net/problem/25304
int cash = int.Parse(Console.ReadLine());
for (int i = int.Parse(Console.ReadLine()); i > 0; i--)
{
    int[] value = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
    cash -= value[0] * value[1];
}
Console.WriteLine(cash == 0 ? "Yes" : "No");